lcm/src/lcm/providers/udpm.rs

```diff
@@ -320,11 +288,14 @@ impl<'a> UdpmProvider<'a> {
         debug!("Binding UDP socket");
         let socket = {
             let inaddr_any = IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0));
-            builder.bind(SocketAddr::new(inaddr_any, port))?
+            builder.bind(SocketAddr::new(inaddr_any, addr.port()))?
```

Comment:
I don't actually know if the port needs to be the specified port here. I was having difficulties with it when I wasn't, but the `test-c-server` doesn't have the same port as the LCM URL.